content/BatchRequestContentCollection.md

```diff
@@ -0,0 +1,61 @@
+- It MUST be possible to configure the maximum number of request objects per `BatchRequestContent` in the `BatchRequestContentCollection` to a value lower than the default value of 20 request objects. This is to enable customized scenarios for developers and provide a means of client side optimization/throttling for workloads that may have lower maximums in batched requests.
+- It MUST possible to use a dictionary of existing unique identifiers and their corresponding to response status code to create a new/fresh `BatchRequestContentCollection` instance to unlock re-emitting/retrying failed requests from an existing `BatchRequestContentCollection` instance.
+
+> Given the similar API experience between `BatchRequestContentCollection` and `BatchRequestContent`, public documentation(code comments and examples) SHOULD encourage the use and point to `BatchRequestContentCollection` to avoid confusion for the end user. Adding new requests should automatically generate new `BatchRequestContent` instances to the `BatchRequestContentCollection`. `BatchRequestContent` should be a public facing type to enable users with customized batching scenarios.
+
+## Performance Considerations
+
+To minimize the risk of clients getting throttled by the API, the request client SHOULD send `BatchRequestContent` instances in sequence if the `BatchRequestContentCollection` contains more than one `BatchRequestContent` instance by default. The request client MAY send `BatchRequestContent` instances in parallel by use of a configuration parameter to allow the user to opt-in to this behavior in the event that the user is aware of the potential throttling limits.
+
+## Usage Examples
+
+```cs
+// create the batch request content collection
+var batchRequestContentCollection = new BatchRequestContentCollection(graphServiceClient);
+var requestInformation = graphServiceClient.Users.ToGetRequestInformation();
+
+// add step using requestInformation
+var requestStepId = await batchRequestContentCollection.AddBatchRequestStepAsync(requestInformation);
+
+// add step using native platform HTTP request
+var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "http://example.com");
+var secondRequestStepId = batchRequestContentCollection.AddBatchRequestStep(httpRequestMessage);
+
+for (var i = 0; i < 25; i++) // its possible to add more than 20 requests
+{
+    BatchRequestStep batchRequestStep = new BatchRequestStep(i.ToString(), new HttpRequestMessage(HttpMethod.Get, REQUEST_URL));
+    batchRequestContentCollection.AddBatchRequestStep(batchRequestStep);
+}
+
+// send and get back response
+var batchResponseContentCollection = await graphServiceClient.Batch.PostAsync(batchRequestContentCollection);
+
+// enumerate list of failed requests and create a new batch request
```

Comment:
I'm unsure about this comment as it seems we're getting the full list of response status codes... Not just the failed ones.